src/components/Body.vue

```diff
@@ -62,27 +62,67 @@ import {
   EMOJI_RESULT_KEY,
   EMOJI_NAME_KEY,
 } from '../constant'
-import { filterEmojis, unicodeToEmoji, isMac } from '../helpers'
+import {
+  filterEmojis,
+  unicodeToEmoji,
+  isMac,
+  sortGroupOrder,
+  snakeToCapitalizedCase,
+} from '../helpers'
 
 export default defineComponent({
   name: 'Body',
-  setup() {
+  props: {
+    additionalGroups: {
+      type: Object,
+      default: () => ({}),
+    },
+    groupOrder: {
+      type: Array,
+      default: () => [],
+    },
+  },
+  setup(props) {
     const { state, updateEmoji, updateSelect } = inject('store') as Store
     const bodyInner = ref<HTMLElement | null>(null)
-    const emojis = computed<EmojiRecord>(() =>
-      filterEmojis(
-        state.emojis,
-        state.search,
-        state.skinTone,
-        state.options.disabledGroups
+    const emojis = computed<EmojiRecord>(() => {
+      const filteredEmojis = Object.entries(
+        filterEmojis(
+          { ...state.emojis, ...props.additionalGroups },
+          state.search,
+          state.skinTone,
+          state.options.disabledGroups
+        )
       )
-    )
+
+      if (props.groupOrder.length) {
+        return Object.fromEntries(
+          filteredEmojis.sort(([a], [b]) =>
+            sortGroupOrder(a, b, props.groupOrder as string[])
+          )
+        )
+      }
+
+      return Object.fromEntries(filteredEmojis)
+    })
 
     const _this = getCurrentInstance()
     const hasGroupNames = computed(() => !state.options.hideGroupNames)
     const isSticky = computed(() => !state.options.disableStickyGroupNames)
     const groupNames = state.options.groupNames
```

Comment:
I think, when you use `groupNames[k] = state.options.groupNames[k]` in line 118, it will try to override the original store directly. Since the store is read-only, it won't work. Maybe we can use something like

`const groupNames = toRaw(state.options.groupNames)`?

Reply:
Introduced in 025275bc4f124aa22df8121192ab0d5b567b9264.